protocol/src/main/java/io/zeebe/protocol/record/value/IncidentRecordValue.java

```diff
@@ -74,9 +74,9 @@
   long getJobKey();
 
   /**
-   * @return the key of the element instance to use in order to update the correct variables before.
-   *     Can be <code>-1</code> if the incident record is part of a {@link IncidentIntent#RESOLVE}
-   *     command.
+   * @return the key of the variable scope on which the variables need to be updated. Can be <code>
+   *     -1</code> if the incident record is part of a {@link IncidentIntent#RESOLVE} command. In
+   *     some cases the returned value is equal to value of {@link #getElementInstanceKey()}.
```

Comment:
I'm not sure if this is more confusing than helpful:

> In some cases the returned value...

We should either explain the cases when this happens or remove this hint :sweat_smile: